packages/core/src/api/logout.ts

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {AuthClient, ResponseMode} from '../auth-client';
+import AsgardeoUIException from '../exception';
+
+/**
+ * Log out the user.
+ *
+ * This function sends a logout request to the server.
+ *
+ * @returns {Promise<void>} A promise that resolves when the logout process is complete.
+ *
+ * @example
+ * logout()
+ *   .then(() => {
+ *     console.log('Logged out!');
+ *   })
+ *   .catch((error) => {
+ *     console.error('Failed to log out:', error);
+ *   });
+ */
+const logout = async (): Promise<void> => {
```

Comment:
Should this be `logout` or `signOut`?
Are we using `signIn` or `Login` in the SDK?

Reply:
We'll be having a component called `SignIn`. (But this is not exposed from this `js-ui-core` package.)
We don't have a separate function called signIn or login. Rather we handle this process using `authorize` and `authenticate` functions which are api calls to `authorize` and `authn` endpoints.

I went with the `logout` name to match the endpoint, `oidc/logout` and the IS & Asgardeo docs also follow the `logout` name. But it seems the currently available js SDKs use the `signOut` more.

Reply:
Updated to `signOut` with 0792b1ffb9cca60c579761932049b07790eb53e6